website/docs/doc-comments.md

```diff
@@ -0,0 +1,134 @@
+---
+id: doc-comments
+title: Documentation Comments
+---
+
+Sorbet's language server has rudimentary support for displaying documentation
+comments in various parts of the editor experience.
+
+For example, hovering over things shows documentation associated with the
+definition:
+
+![](/img/hover-doc-comment.png)
+
+Autocompletion results include documentation for the completion item:
+
+![](/img/autocompletion-doc-comment.png)
+
+## Adding documentation
+
+There is no special syntax for defining documentation comments: Sorbet assumes
+that any Ruby comment immediately before a definition is that definition's
+documentation comment:
+
+```ruby
+# typed: true
+
+# A simple class with some documentation
+class A
+  # The documentation for the A#foo method
+  def foo; end
+end
+
+A.new.foo
+```
+
+To document a method with a signature, put the documentation comment above the
+signature:
+
+```ruby
+# This is the documentation
+sig { void }
+def foo; end
+```
+
+This works for constants as well:
+
+```ruby
+# Documentation for X
+X = 42
+```
+
+### Markdown support
+
+Sorbet's language server will send documentation in Markdown format if the
+language client declares that it supports Markdown (in the `contentFormat` of
+`HoverClientCapabilities` and the `documentationFormat` of
+`CompletionClientCapabilities` in the
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/)).
+VS Code declares that it supports Markdown without configuration required from
+the user, but other language clients may require special configuration.
+
+![](/img/markdown-doc-comment.png)
+
+Markdown support in documentation comments is especially useful for displaying
+examples of usage in code blocks, like seen above.
+
+### Documenting method parameters
+
+At the moment, Sorbet does not support documenting individual method parameters.
+Instead, we recommend documenting parameters in a method's documentation comment
+with the `@param` annotation.
+
+```ruby
+  # @param name Who to greet
+  sig { params(name: String).void }
+  def self.greet(name)
+    puts("Hello, #{name}!")
+  end
+```
+
+Sorbet can generate a YARD snippet with `@param` and `@return` attributes for a
+method, so you don't have to type them all manually:
+
+<video autoplay loop muted playsinline style="max-width: calc(min(813px, 100%));">
+  <source src="/img/lsp/yard-snippet.mp4" type="video/mp4">
```

Comment:
Should the video be committed in this PR? It doesn't seem to be already in the repo: https://github.com/sorbet/sorbet/tree/master/website/static/img

Reply:
Ah I looked at the wrong directory, it's under `lsp/` 👍

Reply:
The video is reused from https://sorbet.org/docs/autocompletion and is present in the `lsp/` subfolder of the folder you linked.